### DISCUSSION TITLE
add a CombineTexture2D

### DISCUSSION THREAD

**Discussion by JB-ER** [godotengine/godot-proposals#11131] [3D]
I had the problem of making a smoke particle system that needs a noise texture, but without noise at the edges of the texture.
You need to make such a texture in an external program right now.

So my idea was to make an CombineTexture2D that takes 2 textures and maybe additionally a mix value and mode (mix, add, subtract, multiply) and combines them into one Texture.
That would be really nice for texture prototyping and keeping your projects small (because you dont need 100 textures for Particles).

**Comment by Jesusemora** [comment]
You can do that with a shader.
A more useful feature would be a built in **method** to blit a shader into a texture without having to make a vulcan program.
Like unity had.

**Reply by JB-ER** [reply]
Yeah shure but then you could just integrate MaterialMaker into Godot (not really, its too complex and big) .
So it would be easier to have one fast and in-editor way of combining textures.
With shaders there would also be the question if they're compile-time calculated or on runtime.
I feel like a CombineTexture would be a good compromise between now (no good texture prototyping, not really complex) and pre-computed shader textures (very variable, too complex).

(A ShaderTexture2D would still be interesting)

**Reply by Jesusemora** [reply]
It's 2 lines of code to combine two textures. What are you talking about?

The only downside of doing mix between two sampler2D is you have two sampler2D. Which is why a blit method would be useful, to bake the result of a material into a texture for 1 frame and then dispose of the material. And yes, this has to be at runtime. Otherwise just do it in gimp.

**Reply by JB-ER** [reply]
You dont get my point. but nvm.
And how do I combine 2 Textures if I'm using a StandardMaterial?

**Reply by Calinou** [reply]
> (A ShaderTexture2D would still be interesting)

This is what https://github.com/godotengine/godot-proposals/issues/7379 is all about :slightly_smiling_face:

**Reply by JB-ER** [reply]
Oh sorry I looked for it but didn't found it, my fault.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/5524.
